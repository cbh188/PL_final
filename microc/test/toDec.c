void main() {
	int a ;
	int b ;
	int c ;
    a = 0b10011010;
	b = 0o567;
	c = 0x127F;
	print a;
	print b;
	print c;
}