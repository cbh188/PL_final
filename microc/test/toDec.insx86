LDARGS 0
CALL (0, "L1_main")
STOP
FLabel (0, "L1_main")
INCSP 1
INCSP 1
INCSP 1
GETBP
OFFSET 0
ADD
CSTI 154
STI
INCSP -1
GETBP
OFFSET 1
ADD
CSTI 375
STI
INCSP -1
GETBP
OFFSET 2
ADD
CSTI 4735
STI
INCSP -1
GETBP
OFFSET 0
ADD
LDI
PRINTI
INCSP -1
GETBP
OFFSET 1
ADD
LDI
PRINTI
INCSP -1
GETBP
OFFSET 2
ADD
LDI
PRINTI
INCSP -1
INCSP -3
RET -1