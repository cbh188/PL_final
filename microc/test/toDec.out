24 19 0 5 25 15 1 15 1 15 1 13 0 0 1 0 154 12 15 -1 13 0 1 1 0 375 12 15 -1 13 0 2 1 0 4735 12 15 -1 13 0 0 1 11 22 15 -1 13 0 1 1 11 22 15 -1 13 0 2 1 11 22 15 -1 15 -3 21 -1